### PR TITLE
chore(query): make row fetcher lazy read segments

### PR DESCRIPTION
**Pull request by sundy-li**
I hereby agree to the terms of the CLA available at: https://databend.rs/dev/policies/cla/

## Summary

Summary about this PR

Closes #issue

### CODE REVIEW

File: src/query/catalog/src/plan/internal_column.rs [hunk 1]
```diff
@@ -60,6 +60,16 @@ pub fn split_row_id(id: u64) -> (u64, u64) {
     (prefix, idx)
 }
 
+pub fn split_prefix(id: u64) -> (u64, u64) {
+    let block_id = id & ((1 << NUM_BLOCK_ID_BITS) - 1);
+
+    let seg_id = id >> NUM_BLOCK_ID_BITS;
+    let seg_id = (seg_id + 1) & ((1 << NUM_SEGMENT_ID_BITS) - 1);
+    let seg_id = (!seg_id).wrapping_add(1);
```
Comment (zhang2014) [inline src/query/catalog/src/plan/internal_column.rs:68]:
```suggestion
    let seg_id = !seg_id.wrapping_sub(1);
```

Comment (RinChanNOWWW) [reply src/query/catalog/src/plan/internal_column.rs:68]: I think this should be 

```rust
let seg_id = id >> NUM_BLOCK_ID_BITS;
let seg_id = seg_id & ((1 << NUM_SEGMENT_ID_BITS) - 1);
let seg_id = !seg_id - 1; // !seg_id + 1 - 1
```

File: src/query/catalog/src/plan/internal_column.rs [hunk 2]
```diff
@@ -60,6 +60,16 @@ pub fn split_row_id(id: u64) -> (u64, u64) {
     (prefix, idx)
 }
 
+pub fn split_prefix(id: u64) -> (u64, u64) {
+    let block_id = id & ((1 << NUM_BLOCK_ID_BITS) - 1);
+
+    let seg_id = id >> NUM_BLOCK_ID_BITS;
+    let seg_id = (seg_id + 1) & ((1 << NUM_SEGMENT_ID_BITS) - 1);
```
Comment (zhang2014) [inline src/query/catalog/src/plan/internal_column.rs:67]:
```suggestion
    let seg_id = (seg_id) | !((1 << NUM_SEGMENT_ID_BITS) - 1);
```

File: src/query/catalog/src/plan/internal_column.rs (hunk 1 above)
Comment (zhang2014) [inline src/query/catalog/src/plan/internal_column.rs:68]:
```suggestion
    let seg_id = (!seg_id).wrapping_add(1) - 1;
```

**Review by RinChanNOWWW — APPROVED** [review]

**Pull request merged**